tf/chunkparser.py

```diff
@@ -309,7 +431,20 @@ def sample_record(self, chunkdata):
                 # Downsample, using only 1/Nth of the items.
                 if random.randint(0, self.sample - 1) != 0:
                     continue  # Skip this record.
+                    
             record = chunkdata[i:i + record_size]
+            
+            # peak at best_q and orig_q from record
+            best_q = struct.unpack('f', record[8284:8288])
+            orig_q = struct.unpack('f', record[8328:8332])
```

Comment:
This code is unsafe here - when upgrading you are attempting to read off the end of the record.  You either need to conditional this logic based on version == V6_VERSION: or move it after appending the additional bytes - or both.

Reply:
ah yes